sql/tsql/TSqlLexer.g4

```diff
@@ -401,6 +401,7 @@ ESTIMATEONLY:                          'ESTIMATEONLY';
 EXCLUSIVE:                             'EXCLUSIVE';
 EXECUTABLE:                            'EXECUTABLE';
 EXIST:                                 'EXIST';
+EXIST_SQUARE_BRACKET:                  '[EXIST]';
```

Comment:
Why is it not possible to use `'[' EXIST ']'` instead of introducing the new token? BTW, the current token doesn't support whitespaces between `[` and `EXIST`. Is it correct?

Reply:
Square brackets are like quotes. Microsoft SSMS doesn't support spaces in this context.
Should we add a test to make sure the grammar doesn't accept it either? Can we do this without new tokens?

![image](https://user-images.githubusercontent.com/168407/217540006-9b258e76-70d6-43f0-8548-7ca416654769.png)

We globally have a problem with brackets, for example with `data_type` rule:

https://github.com/antlr/grammars-v4/blob/bd3d04274567cf4fa86a37fdf99e67fa2724f953/sql/tsql/TSqlParser.g4#L4905-L4912

We added SQUARE_BRACKET_ID here but it really should be `[varchar]`, `[nvarchar]`, `[binary]`, `[varbinary]`.

Reply:
Trying further, square bracketed names do not seem to be possible for built-in functions.

![image](https://user-images.githubusercontent.com/168407/217547428-4836f998-b9e7-4569-b541-ad8d3ced0d9e.png)

The variants introduced in this PR are not for every uses of the non-quoted keywords.

```
KILL QUERY NOTIFICATION SUBSCRIPTION ALL
```
is accepted by SQL Server 2022, but not:
```
KILL [QUERY] NOTIFICATION SUBSCRIPTION ALL
```

Reply:
Actually few can exist with squared brackets. Built-in functions such as ˋrank()` don't, but xml methods do. I can create a generic rule where any bracketed text would be accepted, instead of the four xml methods we currently accept, like we do with data types followed by `(max)`.

Reply:
Ok, let's postpone it.